teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md

```diff
@@ -0,0 +1,565 @@
+    properties:
+    name:
+      type: string
+    digest:
+      type: string
+    digest_ago:
+      type: string
+    prov_type:
+      type: string
+```
+
+``` yaml
+  - name: SIGNABLE-PYTHON_BDIST_$(NAME)
+    type: object
+    description: |
+      * name: name or url of the built distribution
+      * digest: digest of the distribution 
+      * digest_ago: digest algorithm: sha256 etc.
+      * prov_type: 'input' or 'output', indicating provenance type.
+    properties:
+    name:
+      type: string
+    digest:
+      type: string
+    digest_ago:
+      type: string
+    prov_type:
+      type: string
+```
+#### Maven
+
+Maven supports several for compiling and packaging. Compiling is done via `mvn compile` and produces `.class` files and packackging is done via `mvn package`, and [supports](https://maven.apache.org/pom.html#packaging) `pom`, `jar`, `maven-plugin`, `ejb`, `war`, `ear` and `rar`. Some plugins such as [Jib](https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/README.md) can also help containerizing the compiled binaries into OCI images, but that can be covered by OCI image schema.
+
+Therefore the schema for Maven packages is:
+``` yaml
+  - name: SIGNABLE-MAVEN_PACKAGE_$(NAME)
+    type: object
+    description: |
+      * name: name or url of the Maven package. Example: foo-v1.0.jar
+      * digest: digest of the package
+      * digest_ago: digest algorithm: sha256 etc.
+      * prov_type: 'input' or 'output', indicating provenance type.
+    properties:
+    name:
+      type: string
+    digest:
+      type: string
+    digest_ago:
+      type: string
+    prov_type:
+      type: string
+```
+
+#### Go
+
+TODO
+
+
+
+#### NodeJS
+
+TODO
+
+#### Parsing Results from TaskRuns
+
+The defined results are in object types, and will be retrieved from TaskRun results. From the syntax of how the results look, the type of those, whether they are strings, objects or arrays can be determined. We need to validate the targets’ structures against defined result structures. Taking VCS as an example, when a `SIGNABLE-VCS` is found, the target needs to be an object, and it has to have at least the fields in defined `SIGNABLE-VCS` schema: `src_type`, `uri`, `alg` and `digest`. If additional fields are in the target, they will be ignored. The field validation rule follows the same pattern of duck typing mentioned in [TEP-0075](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md). These individual fields will also be validated against supported formats.
+
+#### Storage of Signable Target Provenance Schema
+
+Each target provenance schema will be stored in a file as a serialized json object. They will be included into the Chains controller image. A singleton variable will load them for later use. The naming will be the type hinting marker. For example, VCS schema file will be called `vcs` and it should be stored in the file in such format:
```

Comment:
This is glossing over a lot 😅 

How would provenance be collected from Runs and be provided to Chains? Is https://github.com/tektoncd/community/blob/main/teps/0086-changing-the-way-result-parameters-are-stored.md a prerequisite here, or are you thinking of a different mechanism?

Reply:
I'm removing this section since we are only having one schema, so storage of those schemas may not be needed for now